### PR TITLE
[6.3] Canned Role API remaining tests Automated

### PR DESCRIPTION
**Pull request by jyejare**
Pr includes; 
1. Helper functions to reduce the repetitive code, where the code is prerequisites for test
2. Automated remaining API tests, totaling 17

### CODE REVIEW

File: tests/foreman/api/test_role.py [hunk 1]
```diff
@@ -922,6 +1162,28 @@ def test_negative_assign_org_admin_to_user_group(self):
 
         :CaseLevel: System
         """
+        org_admin = self.create_org_admin_role(
```
[the remark this answers is not in the document]
Comment (rplevka) [reply tests/foreman/api/test_role.py]: @jyejare - bump?

**Comment by san7ket** [comment]
Test Results please :)

File: tests/foreman/api/test_role.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jyejare) [reply tests/foreman/api/test_role.py]: Done

**Comment by jyejare** [comment]
@san7ket , Thanks for asking the results :) because over the period of 3 months since I raised this PR, things changed and hence tests were failing .

I fixed them all now and 3 of them are skipped due to bug I raised yday.

Here are the test results :

```
 py.test -m 'not stubbed' tests/foreman/api/test_role.py::CannedRoleTestCases
=============================================================================================== test session starts ===============================================================================================
platform linux2 -- Python 2.7.15, pytest-3.6.1, py-1.5.4, pluggy-0.6.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare/Desktop/RedHat/RoboTelloNew/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.22.5, services-1.2.1, mock-1.10.0, forked-0.2, cov-2.6.0
collecting 42 items                                                                                                                                                                                               2018-10-10 14:05:59 - conftest - DEBUG - BZ deselect is disabled in settings

collected 42 items / 5 deselected                                                                                                                                                                                 

tests/foreman/api/test_role.py .................ss...s.............                              [100%]

============= 33 passed, 3 skipped, 5 deselected in 868.96 seconds ==============
```

**Pull request merged**